core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -156,8 +158,12 @@ public void put(ClusterService clusterService, PutPipelineRequest request, Actio
             // validates the pipeline and processor configuration before submitting a cluster update task:
             Map<String, Object> pipelineConfig = XContentHelper.convertToMap(request.getSource(), false).v2();
             constructPipeline(request.getId(), pipelineConfig);
+        } catch (ConfigurationPropertyException e) {
```

Comment:
can we move the catching logic to `constructPipeline(...)`? then we can create unit tests for this and remove the java integ tests in `IngestClientIT`.